logger/config.go

```diff
@@ -38,37 +38,45 @@ type Config struct {
 	ItemSampleSeconds  int `yaml:"item_sample_seconds,omitempty"`
 	ItemSampleInitial  int `yaml:"item_sample_initial,omitempty"`
 	ItemSampleInterval int `yaml:"item_sample_interval,omitempty"`
-
-	lock               sync.Mutex       `yaml:"-"`
-	onUpdatedCallbacks []ConfigObserver `yaml:"-"`
 }
 
 type ConfigObserver func(*Config) error
 
+type updatableConfig struct {
+	*Config
+
+	lock               sync.Mutex
+	onUpdatedCallbacks []ConfigObserver
+}
+
+var uConfig updatableConfig
```

Comment:
Don't like the global and having to set the `Config` field in logger.go. Could not think of a better way. Open to suggestions.

Reply:
would prefer to not have the global reference. it can introduce unexpected side-effects for other Config users.

what about changing the Logging field to a pointer type?